.travis.yml

```diff
@@ -5,35 +5,47 @@ env:
 
+services:
+  - docker
+
 # todo: remove this once travis fixes their upstream issue (travis-ci/travis-ci#8836)
 sudo: required
-addons:
-  chrome: stable
 
-script:
-  # Bazel build & test everything. Can't use `bazel build ...` yet, because
-  # javaharness doesn't build on Travis (needs working Android SDK).
-  - >
-    bazel build --noshow_progress --noshow_loading_progress
-    //particles/...
-    //src/...
-    //src_kt/...
-  - >
-    bazel test --noshow_progress --noshow_loading_progress --test_output=errors
-    //particles/...
-    //src/...
-    //src_kt/...
-  - npm run test-with-start
+# Bazel build & test everything. Can't use `bazel build ...` yet, because
+# javaharness doesn't build on Travis (needs working Android SDK).
+jobs:
+  include:
+    - stage: build
+      script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc build //src_kt/..."
+    - stage: test
+      script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc test //src_kt/..."
+      name: "Bazel Test: //src_kt/..."
+    - script: tools/run_in_docker  "bazel --bazelrc=tools/travis.bazelrc test //src/..."
+      name: "Bazel Test: //src/..."
+    - script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc test //particles/..."
+      name: "Bazel Test: //particles/..."
+    - script: tools/run_in_docker "npm run test-no-web"
+      name: "Node Test: Unit"
+#     # TODO(alxr): Add back into CI once docker issue fixed
+#    - script: tools/run_in_docker "npm run test-shells"
+#      name: "Node Test: Shells"
 
+# Use travis to cache a Bazel disk-cache.
+# This connects to the docker container via a volume mount (`-v /home/travis/disk-cache:/disk-cache/`).
+# Bazel will store artifacts here due to the `--disk_cache=/disk-cache/` argument.
 cache:
   npm: true
+  directories:
+    - /home/travis/build/PolymerLabs/arcs/disk-cache/
 
-# Thinking about adding install steps? Why not modify the tools/setup script instead?
-# The list command dumps the top-level packages with their versions, in case we need that for debugging.
-install:
-  - ./tools/setup
-  - npm list --depth=0
+# Pulling container from https://hub.docker.com/repository/docker/alxrsngrtn/arcs-travis-build
+before_install:
+  - docker pull alxrsngrtn/arcs-travis-build:latest
+  - mkdir -p ./disk-cache/
+  - touch .bazelrc
```

Comment:
Use `try-import` instead of `import` in `travis.bazelrc`, and then you can delete this line.

Reply:
Actually, just saw this line in travis:
```
WARNING: Duplicate rc file: /usr/src/app/.bazelrc is read multiple times, most recently imported from /usr/src/app/tools/travis.bazelrc
```

I'm not sure what is going on there. Should we maybe just delete the try-import?

Reply:
Removed import.